scripts/globals/combat/physical_utilities.lua

```diff
@@ -428,6 +428,46 @@ xi.combat.physical.calculateMeleePDIF = function(actor, target, weaponType, wsAt
     return pDif
 end
 
+-- calculates base melee critical rate
+xi.combat.physical.calculateMeleeCriticalRate = function(actor, target, unClamped)
+    -- See reference at https://www.bg-wiki.com/ffxi/Critical_Hit_Rate
+    local finalCriticalRate = 0
+    local baseCriticalRate  = 0.05
+    local dDex              = actor:getStat(xi.mod.DEX) - target:getStat(xi.mod.AGI)
+    local statBonus         = 0
+
+    if dDex > 50 then
+        statBonus = 0.15
+    elseif dDex >= 40 then
+        statBonus = (dDex - 35) / 100
+    elseif dDex >= 30 then
+        statBonus = 0.04
+    elseif dDex >= 20 then
+        criticalRate = 0.03
```

Comment:
error

Reply:
![image](https://github.com/LandSandBoat/server/assets/131182600/ac9846c3-e5a0-4967-a474-9e907aff5ef2)

I'm never copying from your test again

Reply:
I never said I tested it

Reply:
Btw if you squash we lose the co-author commits i cherry-picked

Reply:
WELL THEN! LETS LEARN ADVANCED REBASING! (by "lets" I mean "you people who are not me")

Reply:
...I ***guess*** you could just put all the people as "co authored by" but that's a lot less ~~fun for me to watch~~ _learning a valuable new skill._

Reply:
I actually meant to just `hold` it. Couse now I need to see what we have in weaponskills.lua

Couse I'll be honest, I dont see why we cant just have all logic regarding critical hit rate in 1 function instead of several, even if that means purging whats already in weaponskills.lua.

The only thing that could potentially conflict is the fTP step. Maybe. A mere matter of just passing nill and doing a nill check, maybe. I dunnow, hence why I have to read.

Reply:
Well this PR does remove the crit rate function in ws.lua 

I made two functions because you didn't like the parameter in a single one

And by "you" I mean "you people who are not me"